test/performance/vectorize_pred.cpp

```diff
@@ -16,6 +16,27 @@ const char *string_of_type();
 
 DECL_SOT(float);
 
+template<typename T>
+T tolerance() {
+    return 0;
+}
+
+template<>
+float tolerance<float>() {
+    return 1e-7f;
+}
+
+template<>
+double tolerance<double>() {
+    return 1e-14f;
+}
+
+template<typename T>
+bool equals(T a, T b, T epsilon = tolerance<T>()) {
+    T error = std::abs(a - b);
```

Comment:
I think this needs specializations for float/double, since you want to call `fabs` there.

Reply:
`std::abs()` should have the proper overloads for these types.

Reply:
You're right, this should be fine.